cfe_internal/enterprise/federation/federation.cf

```diff
@@ -399,10 +401,31 @@ bundle agent superhub_schema
                      "cfdb",
                      "'select superhub_schema();'",
                    },
-        classes => superhub_schema_exit_codes;
+        classes => psql_wrapper_exit_codes;
 }
 
-body classes superhub_schema_exit_codes
+bundle agent ensure_feeders
+# @brief Run SQL function to ensure that all configured feeder hubs are in __hubs table
+{
+  vars:
+    am_superhub::
+      "feeders_arg" string => concat( "ARRAY['",
+                                      join( "', '",
+                                            getvalues( "cfengine_enterprise_federation:config.feeder" )
+                                      ),
+                                      "']");
+
+  commands:
+    am_superhub::
+      "$(cfengine_enterprise_federation:config.bin_dir)/psql_wrapper.sh"
+        arglist => {
+                     "cfdb",
+                     "\"select ensure_feeders($(feeders_arg));\""
```

Comment:
I hate having to escape quotes. Consider using single quote or backtick on the outside, then you don't have to https://docs.cfengine.com/docs/3.12/reference-language-concepts-variables.html#quoting